src/Access/AccessControlHandlerInterface.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Drupal\password_enhancements\Access;
+
+use Drupal\Core\Access\AccessResultInterface;
+use Drupal\Core\Routing\Access\AccessInterface;
+use Drupal\Core\Session\AccountInterface;
+
+/**
+ * Defines required methods for the access control handler.
+ */
+interface AccessControlHandlerInterface extends AccessInterface {
```

Comment:
It may be empty but it is used to check whether it is an Access handler or not, see: CheckProvider.php:102

If a class has public methods I think it is a good idea to define interfaces and document their public methods on the interface only, regardless how it is used.

Reply:
1) It is only being checked (still) if it is an access checker service. Yours not anymore.
Proof:
https://github.com/apigee/apigee-edge-drupal/blob/922b1b8c6ec57df3ee8f602b7962cac57b26c82f/apigee_edge.routing.yml#L183
https://github.com/apigee/apigee-edge-drupal/blob/c64b20c5b35774be578adb8a4ef95aa115830d74/src/Controller/ExportAnalyticsController.php#L69

2) You should only define an interface if you are building something that could be decoupled or overridden (in case of a service). The access checker in a route is no such thing, I described earlier why. In this module provide a concrete implementation for checking access on the route that you defined - concrete, not abstract - and customer apps can override it, by overriding the access checker (callback) on the definition route, _but_ the definition of the new access checker class does not need to comply with what you have in the `AccessControlHandlerInterface`, it is not required to define the `hasPasswordChangeAccess` method, it access checker method could by anything. Consequently, the `AccessControlHandlerInterface` is not necessary.

Reply:
Okay.